pkg/api/api.go

```diff
@@ -0,0 +1,400 @@
+type CanaryTestFailInstance struct {
+	Name     string   `json:"name"`
+	Url      string   `json:"url"`
+	PassRate PassRate `json:"passRate"`
+}
+
+// PassRatesByJobName is responsible for the section job pass rates by job name
+type PassRatesByJobName struct {
+	Name      string              `json:"name"`
+	Url       string              `json:"url"`
+	PassRates map[string]PassRate `json:"passRates"`
+}
+
+// FailingTestBug describes a single instance of failed test with bug or failed test without bug
+// differs from failingtest in that it includes pass rates for previous days and latest days
+type FailingTestBug struct {
+	Name      string              `json:"name"`
+	Url       string              `json:"url"`
+	PassRates map[string]PassRate `json:"passRates"`
+	Bugs      []util.Bug          `json:"bugs,omitempty"`
+}
+
+// JobSummaryPlatform describes a single platform and its associated jobs, their pass rates, and failing tests
+type JobSummaryPlatform struct {
+	Platform  string              `json:"platform"`
+	PassRates map[string]PassRate `json:"passRates"`
+}
+
+// FailureGroup describes a single failure group - does not show the associated failed job names
+type FailureGroup struct {
+	Job          string `json:"job"`
+	Url          string `json:"url"`
+	TestFailures int    `json:"testFailures"`
+}
+
+// summary across all job
+func summaryAcrossAllJobs(result, resultPrev map[string]util.SortedAggregateTestResult, endDay int) *SummaryAcrossAllJobs {
+	all := result["all"]
+	allPrev := resultPrev["all"]
+
+	latestDays := fmt.Sprintf("latest%dDays", endDay)
+	prevDays := "prev7Days"
+
+	summary := SummaryAcrossAllJobs{
+		TestExecutions: map[string]int{
+			latestDays: all.Successes + all.Failures,
+			prevDays:   allPrev.Successes + allPrev.Failures,
+		},
+		TestPassPercentage: map[string]float64{
+			latestDays: all.TestPassPercentage,
+			prevDays:   allPrev.TestPassPercentage,
+		},
+	}
+
+	return &summary
+}
+
+// stats on failure groups
+func failureGroups(failureGroups, failureGroupsPrev []util.JobRunResult, endDay int) *FailureGroups {
+
+	_, _, median, medianPrev, avg, avgPrev := util.ComputeFailureGroupStats(failureGroups, failureGroupsPrev)
+
+	latestDays := fmt.Sprintf("latest%dDays", endDay)
+	prevDays := "prev7Days"
+
+	failureGroupStruct := FailureGroups{
+		JobRunsWithFailureGroup: map[string]int{
+			latestDays: len(failureGroups),
+			prevDays:   len(failureGroupsPrev),
+		},
+		AvgFailureGroupSize: map[string]int{
+			latestDays: avg,
+			prevDays:   avgPrev,
+		},
+		MedianFailureGroupSize: map[string]int{
+			latestDays: median,
+			prevDays:   medianPrev,
+		},
+	}
+	return &failureGroupStruct
+}
+
+func summaryJobsByPlatform(report, reportPrev util.TestReport, endDay, jobTestCount int) []JobSummaryPlatform {
+	jobsByPlatform := util.SummarizeJobsByPlatform(report)
+	jobsByPlatformPrev := util.SummarizeJobsByPlatform(reportPrev)
+
+	latestDays := fmt.Sprintf("latest%dDays", endDay)
+	prevDays := "prev7Days"
```

Comment:
just make these `latest` and `prev` so it still make sense if this gets used with custom start/end days later. and makes it easier for a consumer to process the keys.